Terminology doc (#63)
3scale-labs/authorino: docs/terminology.md

# Terminology
Here we define some terms that are used in the project, with the goal of avoiding confusion and facilitating more
accurate conversations related to `Authorino`.

If you see terms used that are not here (or are used in place of terms here) please consider contributing a definition
to this doc with a PR, or modifying the use elsewhere to align with these terms.

## Terms

> * **Access token**: a type of temporary security token, tied to an authenticated identity, that is issued by an auth server under request of registered auth clients of this auth server, and that delegates power of operating on behalf of that identity to a party that presents this token on requests to a resource server, including and especially to the auth client itself; it is typically formatted as a JWT

Reviewer: Maybe instead of...

```suggestion
* **Access token**: a type of temporary security credential, tied to an authenticated identity, that is issued by an auth server under request of registered auth clients of this auth server, and that delegates power of operating on behalf of that identity to a party that presents this token on requests to a resource server, including and especially to the auth client itself; it is typically formatted as a JWT
```

Reviewer: Just to avoid defining it with the same word (token) and also because below we are saying that the AuthN process verifies credentials belonging to an identity.

* **Auth**: usually employed as a short for authentication and authorization (AuthN/AuthZ)
* **Auth client**: an application client that uses an auth server, either to authenticate or to authorize identities who want to consume resources from a resources server; it can represent a user application like a SPA or a mobile app, or a resource server
* **Auth server**: a server where auth clients, users, roles, scopes, resources, policies and permissions are stored and managed
* **Authentication (AuthN)**: the process of verifying that a given credential belongs to a claimed-to-be identity; usually resulting in the issuing of an access token

> * **Authorization (AuthZ)**: the process of grating (or denied) access over a resource to a party based on the set of policies and permissions enforced and

Reviewer: ```suggestion
* **Authorization (AuthZ)**: the process of granting (or denying) access over a resource to a party based on the set of policies and permissions enforced
```

# Terminology
Here we define some terms that are used in the project, with the goal of avoiding confusion and facilitating more
accurate conversations related to `Authorino`.

> If you see terms used that are not here (or are used in place of terms here) please considre contributing a definition

Reviewer 2: ```suggestion
If you see terms used that are not here (or are used in place of terms here) please consider contributing a definition
```
